src/test/java/org/opensearch/ad/AnomalyDetectorRestTestCase.java

```diff
@@ -140,6 +140,11 @@ protected AnomalyDetector createAnomalyDetector(AnomalyDetector detector, Boolea
         Response response = TestHelpers
             .makeRequest(client, "POST", TestHelpers.AD_BASE_DETECTORS_URI, ImmutableMap.of(), TestHelpers.toHttpEntity(detector), null);
         assertEquals("Create anomaly detector failed", RestStatus.CREATED, TestHelpers.restStatus(response));
+        try {
```

Comment:
with sleep the test can still fail.  For example, AnomalyDetectorRestApiIT.testStopNonExistingAdJob has sleep but still fails in the following runs.  Is there any way to add retry?

https://github.com/opensearch-project/anomaly-detection/pull/259/files#diff-2ae99af9d0629658ce7a77190cdf01018d0c9cb5c9841019f4929cb83336e4f1R143

REPRODUCE WITH: ./gradlew ‘:integTest’ --tests “org.opensearch.ad.rest.AnomalyDetectorRestApiIT.testStopNonExistingAdJob” -Dtests.seed=D434080C4D95FDA0 -Dtests.security.manager=false -Dtests.locale=ro -Dtests.timezone=Asia/Vladivostok -Druntime.java=14